tests/request.py

```diff
@@ -10,16 +10,24 @@ def sogs_get(client, url, user):
     return client.get(url, headers=x_sogs_for(user, "GET", url))
 
 
+def sogs_post_raw(client, url, data, user, *, ctype='application/octet-stream', extra_headers=None):
+    """
+    POSTs a test `client` request to `url` with the given `data` as bytes body and X-SOGS-*
+    signature headers signing the request for `user`.
+    """
+    headers = x_sogs_for(user, "POST", url, data)
+    for k in extra_headers or []:
+        if k not in headers:
+            headers[k] = extra_headers[k]
+    return client.post(url, data=data, content_type=ctype, headers=headers)
```

Comment:
```suggestion
    return client.post(url, data=data, content_type=ctype, headers={**headers, **extra_headers})
```
Simplification, and also I don't think the `not in headers` makes sense here: if I give an extra header value that conflicts with one of the X-SOGS headers then I'd expect it to override when the header already exists rather than silently being ignored.